crates/oxc_minifier/src/peephole/substitute_alternate_syntax.rs

```diff
@@ -1346,94 +1277,34 @@ mod test {
-        test("(goog$partial(f,a))(b)", "f(a,b)");
-        test("(goog$partial(f,a,b))(c)", "f(a,b,c)");
-        // Don't rewrite if the bind isn't the immediate call target
-        test_same("(goog$bind(f)).call(g)");
-    }
-
-    #[test]
-    #[ignore]
-    fn test_bind_to_call3() {
-        // TODO(johnlenz): The code generator wraps free calls with (0,...) to
-        // prevent leaking "this", but the parser doesn't unfold it, making a
-        // AST comparison fail.  For now do a string comparison to validate the
-        // correct code is in fact generated.
-        // The FREE call wrapping should be moved out of the code generator
-        // and into a denormalizing pass.
-        // disableCompareAsTree();
-        // retraverseOnChange = true;
-        // late = false;
-
-        test("(goog.bind(f.m))()", "(0,f.m)()");
-        test("(goog.bind(f.m,a))()", "f.m.call(a)");
-
-        test("(goog.bind(f.m))(a)", "(0,f.m)(a)");
-        test("(goog.bind(f.m,a))(b)", "f.m.call(a,b)");
-
-        test("(goog.partial(f.m))()", "(0,f.m)()");
-        test("(goog.partial(f.m,a))()", "(0,f.m)(a)");
-
-        test("(goog.partial(f.m))(a)", "(0,f.m)(a)");
-        test("(goog.partial(f.m,a))(b)", "(0,f.m)(a,b)");
-
-        // Without using type information we don't know "f" is a function.
-        test_same("f.m.bind()()");
-        test_same("f.m.bind(a)()");
-        test_same("f.m.bind()(a)");
-        test_same("f.m.bind(a)(b)");
-
-        // Don't rewrite if the bind isn't the immediate call target
-        test_same("goog.bind(f.m).call(g)");
     }
 
+    // FIXME: the cases commented out can be implemented
     #[test]
-    #[ignore]
     fn test_rotate_associative_operators() {
-        test("a || (b || c); a * (b * c); a | (b | c)", "(a || b) || c; (a * b) * c; (a | b) | c");
-        test_same("a % (b % c); a / (b / c); a - (b - c);");
-        test("a * (b % c);", "b % c * a");
-        test("a * b * (c / d)", "c / d * b * a");
-        test("(a + b) * (c % d)", "c % d * (a + b)");
+        test("a || (b || c)", "(a || b) || c");
+        // float multiplication is not always associative
+        // <https://tc39.es/ecma262/multipage/ecmascript-data-types-and-values.html#sec-numeric-types-number-multiply>
+        test_same("a * (b * c)");
+        // test("a | (b | c)", "(a | b) | c");
+        test_same("a % (b % c)");
+        test_same("a / (b / c)");
+        test_same("a - (b - c);");
+        // test("a * (b % c);", "b % c * a");
+        // test("a * (b / c);", "b / c * a");
+        // cannot transform to `c / d * a * b`
+        test_same("a * b * (c / d)");
+        // test("(a + b) * (c % d)", "c % d * (a + b)");
         test_same("(a / b) * (c % d)");
         test_same("(c = 5) * (c % d)");
-        test("(a + b) * c * (d % e)", "d % e * c * (a + b)");
-        test("!a * c * (d % e)", "d % e * c * !a");
+        // test("!a * c * (d % e)", "d % e * c * !a");
     }
```

Comment:
float multiplication is not always associative so I think the original test case is not always correct. Maybe we can add `unsafe_math` option like terser later on though.